specification/security/resource-manager/Microsoft.Security/preview/2021-01-15-preview/ingestionSettings.json

```diff
@@ -0,0 +1,391 @@
+          },
+          {
+            "$ref": "#/parameters/IngestionSettingName"
+          },
+          {
+            "$ref": "#/parameters/IngestionSetting"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IngestionSetting"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete an ingestion setting for the subscription": {
+            "$ref": "./examples/IngestionSettings/DeleteIngestionSetting_example.json"
+          }
+        },
+        "tags": [
+          "Ingestion Settings"
+        ],
+        "description": "Deletes the ingestion settings for this subscription.",
+        "operationId": "IngestionSettings_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/IngestionSettingName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The resource was deleted successfully"
+          },
+          "204": {
+            "description": "The resource was not found"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/ingestionSettings/{ingestionSettingName}/getToken": {
+      "post": {
+        "x-ms-examples": {
+          "Get a ingestion setting on subscription": {
+            "$ref": "./examples/IngestionSettings/GetTokenIngestionSetting_example.json"
+          }
+        },
+        "tags": [
+          "Ingestion Settings"
+        ],
+        "description": "Returns the token that is used for correlating ingested telemetry with the resources in the subscription.",
+        "operationId": "IngestionSettings_GetToken",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/IngestionSettingName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IngestionSettingWithToken"
```

Comment:
Its misleading here to return a full "ingestionSettings" resource with an extra property in the property bag. This is not a /read on the ingestion settings resource and should not return that resource.  It should only return an object with a single token property or similar